app/population/middleware/set-filter.population.js

```diff
@@ -0,0 +1,18 @@
+function setFilterPopulation(items = [], bodyKey) {
+  return async function buildFilter(req, res, next) {
+    const { locations } = req
+    const locationIds = locations?.join()
+
+    const filter = {
+      'filter[location_type]': 'prison',
```

Comment:
These look like filters that are used directly by the API? What we've tried to do is to keep these definitions within the service rather than in middleware or controllers.

This is so if the filters are updated we can handle that change in one place.

It's tricky to fully understand without the surrounding context but will these views require the filter component we use on other pages to separate state? 

If so, you'll need this middleware. If not, you might be able to get away without it and just pass the location IDs directly to a service method that returns prisons, something like `locationsFreeSpacesService. getPrisonFreeSpaces` and then always apply the location type filter within that service.